crm_partner_assign/models/crm_lead.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Moduon Team S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+from odoo import api, fields, models
+
+
+class CRMLead(models.Model):
+    _inherit = "crm.lead"
+
+    partner_assigned_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Assigned Partner",
+        tracking=True,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```

Comment:
```suggestion
        check_company=True
```

Also add `_check_company_auto = False` as a model attribute.

Reply:
Can't change `_check_company_auto` on `crm.lead` because other fields benefits from it.
Why you want to disable `_check_company_auto`?

Reply:
Ah, sorry, I meant set it to True 🤦🏼‍♂️